cost_calculator/app.py

```diff
@@ -22,50 +23,81 @@
 app.secret_key = os.urandom(24)
 
 
-def k8s_container_resource_requirements(container):
+def k8s_container_resource_requirements(container) -> Dict[str, int]:
+    """
+    returns dict in format:
+     {"req_cpu": 1, "req_mem": 1, "lim_cpu": 2, "lim_mem": 2}
+    """
     try:
         max_cpu = 0
         max_memory = 0
         if not container.resources:
-            return max_cpu, max_memory
+            return {
+                "req_cpu": max_cpu,
+                "req_mem": max_memory,
+                "lim_cpu": max_cpu,
+                "lim_mem": max_memory
+            }
         limits = container.resources.limits
+        max_lim_cpu = 0
+        max_lim_mem = 0
         if limits and limits.get('cpu'):
             cpu = parse_quantity(limits['cpu'])
-            max_cpu = cpu
+            max_lim_cpu = cpu
         if limits and limits.get('memory'):
             memory = parse_quantity(limits['memory'])
-            max_memory = memory
+            max_lim_mem = memory
         requests = container.resources.requests
+        max_req_cpu = 0
+        max_req_mem = 0
         if requests and requests.get('cpu'):
             cpu = parse_quantity(requests['cpu'])
-            max_cpu = max(cpu, max_cpu)
+            max_req_cpu = max(cpu, max_cpu)
         if requests and requests.get('memory'):
             memory = parse_quantity(requests['memory'])
-            max_memory = max(memory, max_memory)
-        return max_cpu, max_memory
+            max_req_mem = max(memory, max_memory)
+        return {
+                "req_cpu": max_req_cpu,
+                "req_mem": max_req_mem,
+                "lim_cpu": max_lim_cpu,
+                "lim_mem": max_lim_mem
+            }
     except Exception:
         logger.exception('Error getting resource requirements for container')
 
 
 def k8s_pod_resource_requirements(pod):
-    max_cpu = 0
-    max_memory = 0
+    max_req_cpu = 0
+    max_lim_cpu = 0
+    max_req_memory = 0
+    max_lim_memory = 0
     if pod.spec.init_containers:
         for container in pod.spec.init_containers:
-            cpu, memory = k8s_container_resource_requirements(container)
-            max_cpu = max(cpu, max_cpu)
-            max_memory = max(memory, max_memory)
-    sum_cpu = 0
-    sum_memory = 0
+            resources = k8s_container_resource_requirements(container)
+            req_cpu, req_mem = resources['req_cpu'], resources['req_mem']
+            lim_cpu, lim_mem = resources['lim_cpu'], resources['lim_mem']
+            max_req_cpu = max(req_cpu, max_req_cpu)
+            max_lim_cpu = max(lim_cpu, max_lim_cpu)
+            max_req_memory = max(req_mem, max_req_memory)
+            max_lim_memory = max(lim_mem, max_lim_memory)
```

Comment:
In `k8s_container_resource_requirements()` it is already ensured that all keys have a zero value if the container does not have the corresponding resource set, so maybe eliminate doing it here again?

Reply:
good point, I tried to follow previous approach on too many levels;)